src/lib.rs

```diff
@@ -523,7 +527,16 @@ impl Serializable<32> for JubJubAffine {
                 let final_x =
                     BlsScalar::conditional_select(&x, &x_negated, flip_sign);
 
-                CtOption::new(JubJubAffine { x: final_x, y }, Choice::from(1u8))
+                // If x == 0, flip_sign == sign_bit. We therefore want to reject
+                // the encoding as non-canonical if all of the
+                // following occur:
+                // - x == 0
+                // - flip_sign == true
+                let x_is_zero = x.ct_eq(&BlsScalar::zero());
+                CtOption::new(
+                    JubJubAffine { x: final_x, y },
+                    !(x_is_zero & flip_sign),
+                )
             }),
```

Comment:
Nit: I would avoid to create `x_negated` and `final_x`, the first can just be `&-x` and the second can be just `x`. That would also simplify the creation of the `JubJubAffine`.
In short, something like:
```rust
                // If x == 0, flip_sign == sign_bit. We therefore want to reject
                // the encoding as non-canonical if all of the
                // following occur:
                // - x == 0
                // - flip_sign == true
                let x_is_zero = x.ct_eq(&BlsScalar::zero());
                // Fix the sign of `x` if necessary
                let flip_sign = Choice::from((x.to_bytes()[0] ^ sign) & 1);
                let x = BlsScalar::conditional_select(&x, &-x, flip_sign);

                CtOption::new(JubJubAffine { x, y }, !(x_is_zero & flip_sign))
```

Reply:
Ye, I like it more!! Done!